.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' ).promises;
+const path = require( 'path' );
+const github = require( '@actions/github' );
+const core = require( '@actions/core' );
+const unzipper = require( 'unzipper' );
+const { formatResultsErrors } = require( 'jest-message-util' );
+
+const TEST_RESULTS_LIST = {
+	open: `<!-- __TEST_RESULTS_LIST__ -->`,
+	close: `<!-- /__TEST_RESULTS_LIST__ -->`,
+};
+const TEST_RESULT = {
+	open: '<!-- __TEST_RESULT__ -->',
+	close: '<!-- /__TEST_RESULT__ -->',
+};
+
+const metaData = {
+	render: ( json ) => `<!-- __META_DATA__:${ JSON.stringify( json ) } -->`,
+	get: ( str ) => {
+		const matched = str.match( /<!-- __META_DATA__:(.*) -->/ );
+		if ( matched ) {
+			return JSON.parse( matched[ 1 ] );
+		}
+	},
+};
+
+( async function run() {
+	const token = core.getInput( 'repo-token', { required: true } );
+	const label = core.getInput( 'label', { required: true } );
+	const artifactName = core.getInput( 'artifact-name', { required: true } );
+
+	const octokit = github.getOctokit( token );
+
+	const flakyTests = await downloadReportFromArtifact(
+		octokit,
+		artifactName
+	);
+
+	if ( ! flakyTests ) {
+		return;
+	}
+
+	const { data: issues } = await octokit.rest.issues.listForRepo( {
+		...github.context.repo,
+		labels: label,
+		per_page: 100,
+	} );
+
+	for ( const flakyTestKey in flakyTests ) {
+		const [ testPath, testTitle ] = getTestPathAndTitle( flakyTestKey );
+		const issueTitle = getIssueTitle( testTitle );
+		const reportedIssue = issues.find(
+			( issue ) => issue.title === issueTitle
+		);
+		const testResults = flakyTests[ flakyTestKey ];
+		const isTrunk = getHeadBranch() === 'trunk';
+
+		if ( reportedIssue ) {
+			let body = reportedIssue.body;
+			const meta = metaData.get( body );
```

Comment:
Should probably guard against `meta` being undefined in case somebody accidentally edits the issue.

Reply:
A lot of things here could break if somebody edits the issue manually 😂 , that's why I added a warning on top of the description, and I expect all members with push access should be able to read it 😅 .